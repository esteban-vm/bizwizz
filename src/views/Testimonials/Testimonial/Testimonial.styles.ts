import clsx from 'clsx'

const styles = {
  wrapper: clsx(
    'm-[3rem_0] flex h-[40rem] w-full flex-col items-start justify-evenly',
    'border-[3px] border-solid border-darkgrey-2 bg-darkgrey-2 p-16 text-left',
    'hover:border-primary'
  ),
  icon: clsx('h-6 w-6 fill-yellow'),
  icon_container: clsx('flex flex-row'),
  content: clsx('text-3xl text-lightgrey'),
  client: clsx('flex items-center gap-10'),
  client_data: clsx('text-2xl'),
  photo: clsx('h-full select-none'),
  photo_container: clsx('h-32 w-32 overflow-hidden rounded-full'),
  name: clsx('font-extrabold text-white'),
  designation: clsx('font-medium text-primary'),
}

export default styles
