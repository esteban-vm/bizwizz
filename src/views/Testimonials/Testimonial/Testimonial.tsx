import type { FC, UUID, ImageProps } from '@/types'
import Image from 'next/image'
import { Icon } from '@/components'
import styles from './Testimonial.styles'

export interface TestimonialProps {
  id: UUID
  name: `${Capitalize<string>} ${Capitalize<string>}`
  designation: Capitalize<string>
  stars: 1 | 2 | 3 | 4 | 5
  photo: ImageProps<`client-${1 | 2 | 3}`>
}

const Testimonial: FC<TestimonialProps> = ({ name, designation, stars, photo, ...props }) => {
  return (
    <article className={styles.wrapper} {...props}>
      <div className={styles.icon_container}>
        {[...Array(5).keys()].map((key) => (
          <Icon key={key} className={styles.icon} name={`star-${key + 1 <= stars ? 'full' : 'empty'}`} />
        ))}
      </div>
      <p className={styles.content}>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Et porro dignissimos quisquam distinctio totam
        obcaecati laudantium accusamus id ut architecto!
      </p>
      <div className={styles.client}>
        <div className={styles.photo_container}>
          <Image alt={`${name}'s photo`} className={styles.photo} {...photo} />
        </div>
        <div className={styles.client_data}>
          <p className={styles.name}>{name}</p>
          <p className={styles.designation}>{designation}</p>
        </div>
      </div>
    </article>
  )
}

export default Testimonial
