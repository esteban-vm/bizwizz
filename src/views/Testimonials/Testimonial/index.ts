export { default, type TestimonialProps } from './Testimonial'
