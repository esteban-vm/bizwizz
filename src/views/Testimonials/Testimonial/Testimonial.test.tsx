import { render, cleanup, within, screen } from '@/tests'
import Testimonial from './Testimonial'
import testimonials from '../Testimonials.data'

describe('Testimonial:', () => {
  let asFragment: () => DocumentFragment
  const [testTestimonial] = testimonials

  beforeEach(() => {
    void ({ asFragment } = render(<Testimonial {...testTestimonial} />))
  })

  afterEach(cleanup)

  it('should render correctly', () => {
    expect(asFragment()).toMatchSnapshot()
  })

  it('should have correct structure', () => {
    const testimonialElem = screen.getByRole('article')
    expect(testimonialElem).toBeInTheDocument()
    expect(testimonialElem).toHaveAttribute('id', testTestimonial.id)
    expect(within(testimonialElem).getAllByRole('img', { name: '' })).toHaveLength(5)
    expect(within(testimonialElem).getByText(/lorem ipsum/i)).toBeInTheDocument()
    expect(within(testimonialElem).getByRole('img', { name: `${testTestimonial.name}'s photo` })).toBeInTheDocument()
    expect(within(testimonialElem).getByText(testTestimonial.name)).toBeInTheDocument()
    expect(within(testimonialElem).getByText(testTestimonial.designation)).toBeInTheDocument()
  })
})
