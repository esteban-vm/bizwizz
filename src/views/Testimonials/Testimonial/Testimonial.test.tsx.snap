// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testimonial: > should render correctly 1`] = `
<DocumentFragment>
  <article
    class="m-[3rem_0] flex h-[40rem] w-full flex-col items-start justify-evenly border-[3px] border-solid border-darkgrey-2 bg-darkgrey-2 p-16 text-left hover:border-primary"
    id="cdbc1f6c-4153-40f2-a00c-e0957fa2812b"
  >
    <div
      class="flex flex-row"
    >
      <svg
        aria-hidden="true"
        class="h-6 w-6 fill-yellow"
        role="img"
      >
        <use
          xlink:href="#icon-star-full"
        />
      </svg>
      <svg
        aria-hidden="true"
        class="h-6 w-6 fill-yellow"
        role="img"
      >
        <use
          xlink:href="#icon-star-full"
        />
      </svg>
      <svg
        aria-hidden="true"
        class="h-6 w-6 fill-yellow"
        role="img"
      >
        <use
          xlink:href="#icon-star-full"
        />
      </svg>
      <svg
        aria-hidden="true"
        class="h-6 w-6 fill-yellow"
        role="img"
      >
        <use
          xlink:href="#icon-star-full"
        />
      </svg>
      <svg
        aria-hidden="true"
        class="h-6 w-6 fill-yellow"
        role="img"
      >
        <use
          xlink:href="#icon-star-empty"
        />
      </svg>
    </div>
    <p
      class="text-3xl text-lightgrey"
    >
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Et porro dignissimos quisquam distinctio totam obcaecati laudantium accusamus id ut architecto!
    </p>
    <div
      class="flex items-center gap-10"
    >
      <div
        class="h-32 w-32 overflow-hidden rounded-full"
      >
        <img
          alt="Paul Rodriguez's photo"
          class="h-full select-none"
          data-nimg="1"
          decoding="async"
          height="536"
          loading="lazy"
          src="/_next/image?url=%2Fimages%2Fclient-1.jpg&w=1080&q=75"
          srcset="/_next/image?url=%2Fimages%2Fclient-1.jpg&w=640&q=75 1x, /_next/image?url=%2Fimages%2Fclient-1.jpg&w=1080&q=75 2x"
          style="color: transparent;"
          width="536"
        />
      </div>
      <div
        class="text-2xl"
      >
        <p
          class="font-extrabold text-white"
        >
          Paul Rodriguez
        </p>
        <p
          class="font-medium text-primary"
        >
          Marketer
        </p>
      </div>
    </div>
  </article>
</DocumentFragment>
`;
